source/main.cc

```diff
@@ -631,12 +649,30 @@ int main (int argc, char *argv[])
                                     "different space dimension is given in the parameter file."));
         }
     }
+  catch (ExceptionBase &exc)
```

Comment:
Isn't this block identical to the next one? That would render it moot because `ExceptionBase` is derived from `std::exception`.